File: frontend/src/types/recurring.ts
import type { ExpenseDate, ExpenseType } from './expense'

export type Frequency = 'monthly' | 'weekly' | 'biweekly'

export interface RecurringExpense {
  template_id: string
  expense_name: string
  amount: number
  category: ExpenseType
  frequency: Frequency
  day_of_month?: number | null
  day_of_week?: number | null
  last_of_month: boolean
  last_reminded?: ExpenseDate | null
  last_user_action?: ExpenseDate | null
  active: boolean
}

export interface PendingExpense {
  pending_id: string
  template_id: string
  expense_name: string
  amount: number
  date: ExpenseDate
  category: ExpenseType
  awaiting_confirmation: boolean
  created_at?: string
}

export interface RecurringListResponse {
  recurring_expenses: RecurringExpense[]
}

export interface PendingListResponse {
  pending_expenses: PendingExpense[]
}

File: ios/BudgetMaster/Models/Recurring.swift
import Foundation

// MARK: - Frequency

public enum Frequency: String, Codable, Sendable {
    case monthly
    case weekly
    case biweekly
    case yearly
}

// MARK: - RecurringExpense

public struct RecurringExpense: Codable, Sendable, Identifiable {
    public var id: String { templateId }

    public let templateId: String
    public let expenseName: String
    public let amount: Double
    public let category: String
    public let frequency: Frequency
    public let dayOfMonth: Int?
    public let dayOfWeek: Int?
    public let monthOfYear: Int?
    public let lastOfMonth: Bool
    public let lastReminded: ExpenseDate?
    public let lastUserAction: ExpenseDate?
    public let active: Bool
}

// MARK: - RecurringListResponse

public struct RecurringListResponse: Codable, Sendable {
    public let recurringExpenses: [RecurringExpense]
}

// MARK: - PendingExpense

public struct PendingExpense: Codable, Sendable, Identifiable {
    public var id: String { pendingId }

    public let pendingId: String
    public let templateId: String
    public let expenseName: String
    public let amount: Double
    public let date: ExpenseDate
    public let category: String
    public let awaitingConfirmation: Bool
    public let createdAt: String?
}

// MARK: - PendingListResponse

public struct PendingListResponse: Codable, Sendable {
    public let pendingExpenses: [PendingExpense]
}

// MARK: - PendingConfirmResponse

public struct PendingConfirmResponse: Codable, Sendable {
    public let success: Bool
    public let expenseId: String
    public let message: String
}
